service/src/main/java/uk/gov/hmcts/reform/fpl/service/casesubmission/CaseSubmissionTemplateDataGenerationService.java

```diff
@@ -0,0 +1,725 @@
+package uk.gov.hmcts.reform.fpl.service.casesubmission;
+
+import lombok.RequiredArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.config.HmctsCourtLookupConfiguration;
+import uk.gov.hmcts.reform.fpl.config.utils.EmergencyProtectionOrderDirectionsType;
+import uk.gov.hmcts.reform.fpl.config.utils.EmergencyProtectionOrderReasonsType;
+import uk.gov.hmcts.reform.fpl.config.utils.EmergencyProtectionOrdersType;
+import uk.gov.hmcts.reform.fpl.enums.OrderType;
+import uk.gov.hmcts.reform.fpl.enums.YesNo;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.Applicant;
+import uk.gov.hmcts.reform.fpl.model.ApplicantParty;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.ChildParty;
+import uk.gov.hmcts.reform.fpl.model.DocmosisAnnexDocuments;
+import uk.gov.hmcts.reform.fpl.model.DocmosisFactorsParenting;
+import uk.gov.hmcts.reform.fpl.model.DocmosisHearing;
+import uk.gov.hmcts.reform.fpl.model.DocmosisHearingPreferences;
+import uk.gov.hmcts.reform.fpl.model.DocmosisInternationalElement;
+import uk.gov.hmcts.reform.fpl.model.DocmosisProceeding;
+import uk.gov.hmcts.reform.fpl.model.DocmosisRisks;
+import uk.gov.hmcts.reform.fpl.model.FactorsParenting;
+import uk.gov.hmcts.reform.fpl.model.Grounds;
+import uk.gov.hmcts.reform.fpl.model.GroundsForEPO;
+import uk.gov.hmcts.reform.fpl.model.Hearing;
+import uk.gov.hmcts.reform.fpl.model.HearingPreferences;
+import uk.gov.hmcts.reform.fpl.model.InternationalElement;
+import uk.gov.hmcts.reform.fpl.model.Orders;
+import uk.gov.hmcts.reform.fpl.model.Other;
+import uk.gov.hmcts.reform.fpl.model.OtherProceeding;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.RespondentParty;
+import uk.gov.hmcts.reform.fpl.model.Risks;
+import uk.gov.hmcts.reform.fpl.model.Solicitor;
+import uk.gov.hmcts.reform.fpl.model.common.DocmosisSocialWorkOther;
+import uk.gov.hmcts.reform.fpl.model.common.Document;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentSocialWorkOther;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.EmailAddress;
+import uk.gov.hmcts.reform.fpl.model.common.Telephone;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisApplicant;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisCaseSubmission;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisChild;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisOtherParty;
+import uk.gov.hmcts.reform.fpl.model.docmosis.DocmosisRespondent;
+import uk.gov.hmcts.reform.fpl.service.DocmosisTemplateDataGeneration;
+import uk.gov.hmcts.reform.fpl.service.UserDetailsService;
+import uk.gov.hmcts.reform.pebble.AgeFilter;
```

Comment:
I feel like we should remove this and implement it else where, with this migration to docmosis we are no longer using the pebble templating engine so it seems strange to then be using filters from it. By removing it we can then also remove the Clock bean/functionality I think as that is the only instance it is used and replace it with time which is a bit nicer to use.